### PR TITLE
Add centered dropdown and dropup options

### LINKED ISSUE #34436: V5 Dropdowns Menu alignment: Add .dropdown-menu-center class
I hope the core team adds the `.dropdown-menu-center` utility class to make the `.dropdown-menu` centered.

Reference : https://getbootstrap.com/docs/5.0/components/dropdowns/#menu-alignment

### PR DESCRIPTION
**Pull request by mdo**
Fixes #34436.

### CODE REVIEW

File: site/content/docs/5.1/examples/carousel/carousel.rtl.css [hunk 1]
```diff
@@ -49,6 +49,10 @@ body {
   margin: 5rem 0; /* Space out the Bootstrap <hr> more */
 }
 
+/* Thin out the marketing headings */
+.featurette-heading {
+}
+
```
Comment (ffoodd) [inline site/content/docs/5.1/examples/carousel/carousel.rtl.css:55]: Seems unrelated, maybe we have a missing step in RTL examples' CSS processing?

**Comment by ffoodd** [comment]
Unrelated changes apart, it works great!

**Comment by mdo** [comment]
Pulled out the unrelated change, thanks for catching! Should be good to approve now.

**Review by ffoodd — APPROVED** [review]
:shipit:

File: js/src/dropdown.js [hunk 2]
```diff
@@ -248,6 +252,14 @@ class Dropdown extends BaseComponent {
       return PLACEMENT_LEFT
     }
 
+    if (parentDropdown.classList.contains(CLASS_NAME_DROPUP_CENTER)) {
```
Comment (XhmikosR) [inline js/src/dropdown.js:255]: These need a couple of tests
